src/main.rs

```diff
@@ -69,29 +77,73 @@ impl Client {
     // Method should return an error when a connection already exists.
     // The client should send a handshake to the server.
     fn open(&mut self, addr: &str, server: Server) -> CommsResult<()> {
-        todo!()
+        if let std::collections::hash_map::Entry::Vacant(e) =
+            self.connections.entry(String::from(addr))
+        {
+            e.insert(Connection::Open(server));
+            let res = self.send(addr, Message::new(MessageType::Handshake, self.ip.clone()));
+            match res {
+                Err(x) => Err(x),
+                _ => Ok(()),
+            }
```

Comment:
Można skorzystać z kombinatora `and()` lub `map()`:
1. `self.send().and( Ok(()) )`
2. `self.send().map(|_| ())`